### PR TITLE
[CI] Automate moving of bot PRs to board

### PR DESCRIPTION
**Pull request by surchs**
Closes https://github.com/neurobagel/workflows/issues/24

### CODE REVIEW

File: .github/workflows/global_move_bot_pr_to_board.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+          MAX_REPO: 50
+        run: |
+          pr_urls=$(gh search prs --owner neurobagel label:_bot --state open --json 'url' | jq -c '[.[].url]')
+          echo "pr_urls=${pr_urls}" >> $GITHUB_OUTPUT
+
+  process_prs:
+    name: Process PR
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.NB_PROJECT_PAT }}
```
Comment (alyssadai) [inline .github/workflows/global_move_bot_pr_to_board.yml:36]: Interesting. Maybe this is what you were saying in the issue description, but does this mean that the PAT can't be the same for the `get_prs` job and the `process_prs` job?

File: .github/workflows/add_pr2project.yml [hunk 2]
```diff
@@ -1,155 +0,0 @@
...
-        continue-on-error: true
-
-      - name: Add PR to Project
-        if: steps.in_project.outcome == 'failure' && steps.preview_label_check.outputs.label_check == 'success' 
-        uses: actions/add-to-project@main
-        with:
-          project-url: https://github.com/orgs/neurobagel/projects/1
-          github-token: ${{ secrets.NB_PROJECT_PAT }}
-
-        # Note that in contrast to the graphical github UI,
```
Comment (alyssadai) [inline .github/workflows/add_pr2project.yml:43]: I know you refactored the workflow but I still found these comments very useful for understanding what's going on in the `Find project node id` and `Get IDs for Status field and Community option` steps of the `process_prs` job in the new yml below.

Could we add these comments back into the new workflow?

File: .github/workflows/add_pr2project.yml [hunk 3]
```diff
@@ -1,155 +0,0 @@
...
-                  # We have passed through the entire list of items 
-                  # and didn't find the project card for our PR.
-                  # Something is wrong and we will now crash the workflow.
-                  exit 1;
-              fi
-          done
-          )
-          echo "parent_id=${out}" >> $GITHUB_OUTPUT
-
-        # This step expects a custom field called "Status"
```
Comment (alyssadai) [inline .github/workflows/add_pr2project.yml:109]: Same comment as above - I thought this comment was very helpful!

**Review by alyssadai — APPROVED** [review]
Thanks @surchs for fixing this! I think the overall logic makes sense to me.

As always, awed by your handle on this GitHub automation and API stuff...

A few clarification comments below, otherwise 🧑‍🍳 ! Also, don't forget to add a PR prefix, hehe!🙂

File: .github/workflows/global_move_bot_pr_to_board.yml (hunk 1 above)
Comment (surchs) [reply .github/workflows/global_move_bot_pr_to_board.yml:36]: Yes, I'll add a comment

**Pull request merged**